### PR TITLE
Adding unsubscribe method to observer

### PR DESCRIPTION
**Pull request by PythooonUser**
## Summary
This adds an unsubscribe method to the observer.

Relates to issue #26.

### CODE REVIEW

File: twitchobserver/twitchobserver.py [hunk 1]
```diff
@@ -122,6 +122,12 @@ def subscribe(self, callback):
 
         self._subscribers.append(callback)
 
+    def unsubscribe(self, callback):
+        """Unsubscribe a callback from the observer."""
+
+        if callback in self._subscribers:
+            del self._subscribers[self._subscribers.index(callback)]
```
Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: I'm curious why you are using ```del list[index]``` over ```list.remove(index)```.

Comment (JoshuaSkelly) [inline twitchobserver/twitchobserver.py]: I think it is a bit simpler(and I do prefer more simple 😄 ). Otherwise I do believe it is functionally the same.

Comment (PythooonUser) [inline twitchobserver/twitchobserver.py]: Alright! :)

**Comment by PythooonUser** [comment]
That might be even more simpler, without using the index but the object directly.

**Comment by JoshuaSkelly** [comment]
Looks good! 👍  Feel free to commit when you are ready.

**Comment by PythooonUser** [comment]
Closes #26.

**Pull request merged**